presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+        OGCGeometry bound = geometry.boundary();
+        if (bound.geometryType().equals(MULTI_LINE_STRING) && ((OGCMultiLineString) bound).numGeometries() == 1) {
+            bound = ((OGCMultiLineString) bound).geometryN(0);
+        }
+        return serialize(bound);
+    }
+
+    @Description("Returns a geometry representing the double precision (float8) bounding box of the supplied geometry")
+    @ScalarFunction("ST_Envelope")
+    @SqlType(NAME)
+    public static Slice stEnvelope(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        Envelope envelope = getEnvelope(geometry);
+        return serialize(OGCGeometry.createFromEsriGeometry(envelope, reference));
+    }
+
+    @Description("Returns a geometry that represents that part of geometry A that does not intersect with geometry B")
+    @ScalarFunction("ST_Difference")
+    @SqlType(NAME)
+    public static Slice stDifference(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return serialize(leftGeometry.difference(rightGeometry));
+    }
+
+    @Description("For geometry type Returns the 2-dimensional cartesian minimum distance (based on spatial ref) between two geometries in projected units")
+    @ScalarFunction("ST_Distance")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stDistance(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return leftGeometry.distance(rightGeometry);
+    }
+
+    @Description("Returns a line string representing the exterior ring of the POLYGON geometry")
+    @ScalarFunction("ST_ExteriorRing")
+    @SqlType(NAME)
+    public static Slice stExteriorRing(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_ExteriorRing", POLYGON);
+        try {
+            Method method = OGCPolygon.class.getMethod("exteriorRing");
+            OGCLineString exteriorRing = (OGCLineString) method.invoke(geometry);
+            return serialize(exteriorRing);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Description("Returns a geometry that represents the shared portion of geometry A and geometry B")
+    @ScalarFunction("ST_Intersection")
+    @SqlType(NAME)
+    public static Slice stIntersection(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return serialize(leftGeometry.intersection(rightGeometry));
+    }
+
+    @Description("Returns a geometry that represents the portions of A and B that do not intersect")
```

Comment:
nit: A & B vs. left & right

perhaps, "Returns the Geometry value that represents the point set symmetric difference of two geometries"